File: backend/HomomorphicEncryptionBackend/Resolver/StageResolver.py
"""
HomomorphicEncryptionBackend.Resolver.StageResolver
"""

from injector import inject
from HomomorphicEncryptionBackend.Resolver.StepResolver import StepResolver
from HomomorphicEncryptionBackend.Model.Stage import Stage

__author__ = "VJ Patel (vj@vjpatel.me)"


class StageResolver:
    """
    StageResolver
    """

    @inject(step_resolver=StepResolver)
    def __init__(self, step_resolver):
        """
        :return:
        """
        self.__step_resolver = step_resolver

    def from_dict(self, stage_dict):
        stage = Stage()

        stage.set_name(stage_dict['name'])

        for step_dict in stage_dict['steps']:
            stage.add_step(
                self.__step_resolver.from_dict(step_dict)
            )

        return stage

File: client/src/app/resolver/stage_resolver.ts
import {Injectable} from "angular2/core";
import {Stage} from "../model/stage";
import {StepResolver} from "./step_resolver";

@Injectable()
export class StageResolver {

  private stepResolver: StepResolver;

  constructor(
    stepResolver: StepResolver
  ) {
    this.stepResolver = stepResolver;
  }

  public fromJson(jsonArr, stage?: Stage): Stage {
    if (!stage) {
      stage = new Stage();
    }
    stage
      .setName(jsonArr.name)
      .setHost(jsonArr.host)
    ;

    jsonArr.steps.forEach(stepJson => {
      let step = this.stepResolver.fromJson(stepJson);
      stage.addStep(step);
    });

    return stage;
  }
}
